### PR TITLE
Unit testing data types

### LINKED ISSUE #4930: Docs for unit testing, data types
### Contributions

- [X] I have read the contribution docs, and understand what's expected of me.

### Link to the page on docs.getdbt.com requiring updates

https://docs.getdbt.com/reference/resource-properties/unit-tests

### What part(s) of the page would you like to see updated?

New section for data types 

Similar to https://docs.getdbt.com/reference/resource-properties/constraints#platform-specific-support

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by matthewshaver**
## What are you changing in this pull request and why?

Adds a section for platform specific data types.

Also took the opportunity to remove the `versions` section of the unit testing doc as it was redundant.

## Checklist

- [ ] Review the [Content style guide](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/content-style-guide.md) so my content adheres to these guidelines.
- [ ] For [docs versioning](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/single-sourcing-content.md#about-versioning), review how to [version a whole page](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/single-sourcing-content.md#adding-a-new-version) and [version a block of content](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/single-sourcing-content.md#versioning-blocks-of-content).
- [ ] Add a checklist item for anything that needs to happen before this PR is merged, such as "needs technical review" or "change base branch."

Adding or removing pages (delete if not applicable):
- [ ] Add/remove page in `website/sidebars.js`
- [ ] Provide a unique filename for new pages
- [ ] Add an entry for deleted pages in `website/vercel.json`
- [ ] Run link testing locally with `npm run build` to update the links that point to deleted pages

### CODE REVIEW

File: website/docs/reference/resource-properties/data-types.md [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+           timestamp_field: 2013-11-03 00:00:00-0
+           timestamptz_field: 2013-11-03 00:00:00-0
+           number_field: 3
+           variant_field: 3
+           geometry_field: POINT(1820.12 890.56)
+           geography_field: POINT(-122.35 37.55)
+           object_field: {'Alberta':'Edmonton','Manitoba':'Winnipeg'}
+           str_array_field: ['a','b','c']
+           int_array_field: [1, 2, 3]
+		...
```
Comment (matthewshaver) [inline website/docs/reference/resource-properties/data-types.md]:
```suggestion

```

File: website/docs/reference/resource-properties/data-types.md [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+           bigint_field: 1
+           geography_field: 'st_geogpoint(75, 45)'
+           json_field: {"name": "Cooper", "forname": "Alice"}
+           str_array_field: ['a','b','c']
+           int_array_field: [1, 2, 3]
+           date_array_field: ['2020-01-01']
+           struct_field: 'struct("Isha" as name, 22 as age)'
+           struct_of_struct_field: 'struct(struct(1 as id, "blue" as color) as my_struct)'
+           struct_array_field: ['struct(st_geogpoint(75, 45) as my_point)', 'struct(st_geogpoint(75, 35) as my_point)']
+		...
```
Comment (matthewshaver) [inline website/docs/reference/resource-properties/data-types.md]:
```suggestion

```

File: website/docs/reference/resource-properties/data-types.md [hunk 3]
```diff
@@ -0,0 +1,151 @@
+---
+title: "Platform-specific data types"
+sidebar_label: "Data types"
+---
+
+Unit tests are designed to test for the expected _values_, not for the data types themselves. dbt takes the value you provide and attempts to cast it to the data type as inferred from the input and output models. 
+
+How you specify input and expected values in your unit test YML definitions are largely consistent across data warehouses, with some variation for more complex data types.
```
Comment (matthewshaver) [inline website/docs/reference/resource-properties/data-types.md]:
```suggestion
How you specify input and expected values in your unit test YML definitions are largely consistent across data warehouses, with some variation for more complex data types. The following are platform-specific data types:
```

File: website/docs/reference/resource-properties/data-types.md [hunk 4]
```diff
@@ -0,0 +1,151 @@
+---
+title: "Platform-specific data types"
+sidebar_label: "Data types"
+---
+
+Unit tests are designed to test for the expected _values_, not for the data types themselves. dbt takes the value you provide and attempts to cast it to the data type as inferred from the input and output models. 
+
+How you specify input and expected values in your unit test YML definitions are largely consistent across data warehouses, with some variation for more complex data types. The following are platform-specific data types:
```
Comment (nghi-ly) [inline website/docs/reference/resource-properties/data-types.md]:
```suggestion
How you specify input and expected values in your unit test YAML definitions are largely consistent across data warehouses, with some variation for more complex data types. The following are platform-specific data types:
```

File: website/docs/reference/resource-properties/data-types.md [hunk 5]
```diff
@@ -0,0 +1,151 @@
...
+           str_escaped_field: "my,cool'string"
+           bool_field: true
+           date_field: 2020-01-02
+           timestamp_field: 2013-11-03 00:00:00-0
+           timestamptz_field: 2013-11-03 00:00:00-0
+           json_field: '{"bar": "baz", "balance": 7.77, "active": false}'
+
+```
+
+Note that `array` is not supported at this time.
```
Comment (nghi-ly) [inline website/docs/reference/resource-properties/data-types.md]:
```suggestion
Currently, the `array` key is not supported at this time.
```

**Review by nghi-ly — COMMENTED** [review]
very easy to read! left a couple of minor nits

File: website/docs/reference/resource-properties/data-types.md (hunk 5 above)
Comment (matthewshaver) [inline website/docs/reference/resource-properties/data-types.md]:
```suggestion
Currently, the `array` key is not supported.
```

File: website/docs/reference/resource-properties/data-types.md [hunk 6]
```diff
@@ -0,0 +1,151 @@
...
+           str_field: my_string
+           str_escaped_field: "my,cool'string"
+           date_field: 2020-01-02
+           timestamp_field: 2013-11-03 00:00:00-0
+           timestamptz_field: 2013-11-03 00:00:00-0
+           json_field: '{"bar": "baz", "balance": 7.77, "active": false}'
+
+```
+
+Note that `array` is not supported at this time.
```
Comment (matthewshaver) [inline website/docs/reference/resource-properties/data-types.md]:
```suggestion
Currently, the `array` is not supported.
```

**Review by nghi-ly — APPROVED** [review]
:shipit:

**Review by MichelleArk — APPROVED** [review]

**Pull request merged**